src/main/test/surfrdUtils_test/test_surfrdUtils.pf

```diff
@@ -112,6 +116,64 @@ contains
      call pftcon%clean()
   end subroutine test_collapse_crop_types_16_to_16
 
+  @Test
+  subroutine test_collapse_crop_types_18_to_16()
+     ! This test starts with two non-generic and two generic crops in the data
+     ! but ends up with only two generic crops because use_crop = .false. and
+     ! irrigate = .true.
+     use pftconMod, only : pftcon, nc3crop, nc3irrig
+     use clm_varctl, only : irrigate, use_crop
+     use clm_varpar, only : maxveg, cft_lb, cft_ub, natpft_size, natpft_lb, natpft_ub
+
+     implicit none
+     integer, parameter :: begg = 2, endg = 3, cftsize = 4
+     integer :: m
+     real(r8) :: wt_cft(begg:endg,cftsize)
+     real(r8) :: fert_cft(begg:endg,cftsize)
+
+     ! Set relevant pftcon values to defaults; override where necessary
+     call pftcon%InitForTesting()
+     wt_cft(begg:,1) = 15.0_r8
+     wt_cft(begg:,2) = 20.0_r8
+     wt_cft(begg:,3) = 30.0_r8
+     wt_cft(begg:,4) = 35.0_r8
+     call check_sums_equal_1( (wt_cft/100.0_r8), begg, "test_check_sums_add_to_1", &
+                              "should not trigger an error")
+     fert_cft(begg:,1) = 15.0_r8  ! these values do not matter; just make sure
+     fert_cft(begg:,2) = 20.0_r8
+     fert_cft(begg:,3) = 30.0_r8
+     fert_cft(begg:,4) = 35.0_r8  ! that they come back the same in this test
+     nc3crop     = 15
+     nc3irrig    = nc3crop + 1
+     do m = 1, nc3irrig
+        pftcon%mergetoclmpft(m) = m
+     end do
+     pftcon%mergetoclmpft(17) = 15
+     pftcon%mergetoclmpft(18) = 16
+     irrigate = .true.
+     use_crop = .false.
+     maxveg = 18  ! # of patches without bare ground
+     natpft_size = maxveg + 1 - cftsize
+     natpft_lb = 0
+     natpft_ub = natpft_lb + natpft_size - 1
+     cft_lb = natpft_ub + 1
+     cft_ub = cft_lb + cftsize - 1
+
+     ! Collapse crop types
+     call collapse_crop_types( wt_cft, fert_cft, cftsize, begg, endg, verbose = .true., sumto = 100._r8 )
+
+     ! Now check that are correct
+     call check_sums_equal_1( wt_cft/100.0_r8, begg, "test_check_sums_add_to_1", &
+                              "should not trigger an error")
+     write(*,*) 'EDW!!', wt_cft, fert_cft
```

Comment:
forgot diagnostic write statement; will remove in next commit/push...

Reply:
I see this one was resolved.